tasks/updateconstants.js

```diff
@@ -258,3 +303,63 @@ function expandItemGroup(key, items) {
 function replaceUselessDecimals(strToReplace) {
   return strToReplace.replace(/\.0+(\D)/, '$1');
 }
+
+// Formats something like "20 21 22" or [ 20, 21, 22 ] to be "20 / 21 / 22"
+function formatValues(value, percent=false, separator=" / ") {
+  var values = Array.isArray(value) ? value : String(value).split(" ");
+  if (values.every(v => v == values[0])) {
+    values = [ values[0] ];
+  }
+  if(percent){
+    values = values.map(v => v + "%");
+  }
+  return values.join(separator).replace(/\.0+(\D|$)/g, '$1');
+}
+
+// Formats AbilitySpecial for the attrib value for abilities and items
+function formatAttrib(attributes, strings, strings_prefix) {
+  return (attributes || []).map(attr => {
+    let key = Object.keys(attr).find(key => `${strings_prefix}${key}` in strings);
+    if (!key){
+      return null;
+    }
+    let header = strings[`${strings_prefix}${key}`];
+    let match = header.match(/(%)?(\+\$)?(.*)/);
+    header = match[3];
+    if (match[2]) {
+      let val = `${match[2][0]} <span class=\"attribVal\">${formatValues(attr[key], match[1])}</span>`;
```

Comment:
Yeah I was thinking about that, seems dotaconstants should just be providing the data instead of making it into html. I could start this by adding a couple more values to hold the data (like add mana and cooldown to abilities for example) but leave the existing values (cmb) for now, so any existing ui code isn't broken in the meantime?